pkg/cmd/ssh/ssh_patch_options.go

```diff
@@ -0,0 +1,343 @@
+package ssh
+
+import (
+	"context"
+	"crypto/x509"
+	"encoding/pem"
+	"errors"
+	"fmt"
+	"net"
+	"sort"
+	"time"
+
+	"github.com/gardener/gardener/pkg/apis/core/v1beta1"
+	operationsv1alpha1 "github.com/gardener/gardener/pkg/apis/operations/v1alpha1"
+	"github.com/spf13/cobra"
+	networkingv1 "k8s.io/api/networking/v1"
+	"k8s.io/client-go/tools/clientcmd/api"
+
+	gardenClient "github.com/gardener/gardenctl-v2/internal/gardenclient"
+	"github.com/gardener/gardenctl-v2/internal/util"
+	"github.com/gardener/gardenctl-v2/pkg/cmd/base"
+	"github.com/gardener/gardenctl-v2/pkg/target"
+)
+
+// wrappers used for unit tests only
+var (
+	getCurrentUser = func(ctx context.Context, gardenClient gardenClient.Client, authInfo *api.AuthInfo) (string, error) {
+		if len(authInfo.ClientCertificateData) > 0 {
+			block, _ := pem.Decode([]byte(authInfo.ClientCertificateData)) // does not return an error, just nil
+			if block == nil {
+				return "", fmt.Errorf("Could not decode PEM certificate")
+			}
+
+			cert, err := x509.ParseCertificate(block.Bytes)
+			if err != nil {
+				return "", err
+			}
+
+			userName := cert.Subject.CommonName
+			if len(userName) > 0 {
+				return userName, nil
+			}
+		}
+
+		if len(authInfo.Token) > 0 {
+			tokenReview, err := gardenClient.CreateTokenReview(ctx, authInfo.Token)
+			return tokenReview.Status.User.Username, err
+		}
+
+		return "", nil
+	}
+)
+
+//nolint:revive
+type SSHPatchOptions struct {
+	sshBaseOptions
+	// the patch target
```

Comment:
```suggestion
	// BastionName is the patch target
```
doc comments for types start with naming the declared symbol